### ISSUE TITLE
Flannel interface down while using with kubeadm

### ISSUE THREAD

**Issue by vjdhama** [coreos/flannel#688]
I ran into the same situation as described in #676. Then as suggested on kubernetes slack, I ran
`kubectl apply -f https://github.com/coreos/flannel/blob/master/Documentation/kube-flannel-rbac.yml`

This fixed the issue of pods not coming up, but flannel interface is still down and pod throws the following error.

```
E0419 17:07:44.681897       1 network.go:102] failed to register network: failed to acquire lease: the server does not allow access to the requested resource (put nodes ubuntu)
E0419 17:07:45.684478       1 network.go:102] failed to register network: failed to acquire lease: the server does not allow access to the requested resource (put nodes ubuntu)
E0419 17:07:46.686523       1 network.go:102] failed to register network: failed to acquire lease: the server does not allow access to the requested resource (put nodes ubuntu)
E0419 17:07:47.688430       1 network.go:102] failed to register network: failed to acquire lease: the server does not allow access to the requested resource (put nodes ubuntu)
E0419 17:07:48.693074       1 network.go:102] failed to register network: failed to acquire lease: the server does not allow access to the requested resource (put nodes ubuntu)
E0419 17:07:49.698987       1 network.go:102] failed to register network: failed to acquire lease: the server does not allow access to the requested resource (put nodes ubuntu)
```

 > kubectl describe po kube-flannel-ds-3v4k4 -n kube-system

  ```
Name:		kube-flannel-ds-3v4k4
Namespace:	kube-system
Node:		ubuntu/10.0.11.10
Start Time:	Wed, 19 Apr 2017 23:55:48 +0700
Labels:		app=flannel
		pod-template-generation=1
		tier=node
Annotations:	kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"DaemonSet","namespace":"kube-system","name":"kube-flannel-ds","uid":"0979f02c-2521-11e7-8948-525400fb8bd2"...
Status:		Running
IP:		10.0.11.10
Controllers:	DaemonSet/kube-flannel-ds
Containers:
  kube-flannel:
    Container ID:	docker://03f1ef629e80d05ffb41f257c704e96a959309da5cda1d31ec013df7ce6a7c61
    Image:		quay.io/coreos/flannel:v0.7.0-amd64
    Image ID:		docker-pullable://quay.io/coreos/flannel@sha256:689be113b2e7803514bf123d63d0fb41d347213b1fe12de7ddfe373281bb98cc
    Port:
    Command:
      /opt/bin/flanneld
      --ip-masq
      --kube-subnet-mgr
    State:		Running
      Started:		Wed, 19 Apr 2017 23:55:48 +0700
    Ready:		True
    Restart Count:	0
    Environment:
      POD_NAME:		kube-flannel-ds-3v4k4 (v1:metadata.name)
      POD_NAMESPACE:	kube-system (v1:metadata.namespace)
    Mounts:
      /etc/kube-flannel/ from flannel-cfg (rw)
      /run from run (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from flannel-token-kj67g (ro)
  install-cni:
    Container ID:	docker://32cba458841c4c7227b2ceb8f3d62f31112cd4db7218521702e8e6b1526c6d3a
    Image:		quay.io/coreos/flannel:v0.7.0-amd64
    Image ID:		docker-pullable://quay.io/coreos/flannel@sha256:689be113b2e7803514bf123d63d0fb41d347213b1fe12de7ddfe373281bb98cc
    Port:
    Command:
      /bin/sh
      -c
      set -e -x; cp -f /etc/kube-flannel/cni-conf.json /etc/cni/net.d/10-flannel.conf; while true; do sleep 3600; done
    State:		Running
      Started:		Wed, 19 Apr 2017 23:55:49 +0700
    Ready:		True
    Restart Count:	0
    Environment:	<none>
    Mounts:
      /etc/cni/net.d from cni (rw)
      /etc/kube-flannel/ from flannel-cfg (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from flannel-token-kj67g (ro)
Conditions:
  Type		Status
  Initialized 	True
  Ready 	True
  PodScheduled 	True
Volumes:
  run:
    Type:	HostPath (bare host directory volume)
    Path:	/run
  cni:
    Type:	HostPath (bare host directory volume)
    Path:	/etc/cni/net.d
  flannel-cfg:
    Type:	ConfigMap (a volume populated by a ConfigMap)
    Name:	kube-flannel-cfg
    Optional:	false
  flannel-token-kj67g:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	flannel-token-kj67g
    Optional:	false
QoS Class:	BestEffort
Node-Selectors:	beta.kubernetes.io/arch=amd64
Tolerations:	node-role.kubernetes.io/master=:Exists:NoSchedule
		node.alpha.kubernetes.io/notReady=:Exists:NoExecute for 300s
		node.alpha.kubernetes.io/unreachable=:Exists:NoExecute for 300s
Events:
  FirstSeen	LastSeen	Count	From		SubObjectPath			Type		Reason	Message
  ---------	--------	-----	----		-------------			--------	------	-------
  12m		12m		2	kubelet, ubuntu	spec.containers{kube-flannel}	Normal		Pulled	Container image "quay.io/coreos/flannel:v0.7.0-amd64" already present on machine
  12m		12m		1	kubelet, ubuntu	spec.containers{kube-flannel}	Normal		Created	Created container with id 03f1ef629e80d05ffb41f257c704e96a959309da5cda1d31ec013df7ce6a7c61
  12m		12m		1	kubelet, ubuntu	spec.containers{kube-flannel}	Normal		Started	Started container with id 03f1ef629e80d05ffb41f257c704e96a959309da5cda1d31ec013df7ce6a7c61
  12m		12m		1	kubelet, ubuntu	spec.containers{install-cni}	Normal		Created	Created container with id 32cba458841c4c7227b2ceb8f3d62f31112cd4db7218521702e8e6b1526c6d3a
  12m		12m		1	kubelet, ubuntu	spec.containers{install-cni}	Normal		Started	Started container with id 32cba458841c4c7227b2ceb8f3d62f31112cd4db7218521702e8e6b1526c6d3a
```
Flannel Manifests

```
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: flannel
---
kind: ConfigMap
apiVersion: v1
metadata:
  name: kube-flannel-cfg
  labels:
    tier: node
    app: flannel
data:
  cni-conf.json: |
    {
      "name": "cbr0",
      "type": "flannel",
      "delegate": {
        "isDefaultGateway": true
      }
    }
  net-conf.json: |
    {
      "Network": "10.244.0.0/16",
      "Backend": {
        "Type": "udp"
      }
    }
---
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  name: kube-flannel-ds
  labels:
    tier: node
    app: flannel
spec:
  template:
    metadata:
      labels:
        tier: node
        app: flannel
    spec:
      hostNetwork: true
      nodeSelector:
        beta.kubernetes.io/arch: amd64
      serviceAccountName: flannel
      containers:
      - name: kube-flannel
        image: quay.io/coreos/flannel:v0.7.0-amd64
        command: [ "/opt/bin/flanneld", "--ip-masq", "--kube-subnet-mgr" ]
        securityContext:
          privileged: true
        env:
        - name: POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: POD_NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        volumeMounts:
        - name: run
          mountPath: /run
        - name: flannel-cfg
          mountPath: /etc/kube-flannel/
      - name: install-cni
        image: quay.io/coreos/flannel:v0.7.0-amd64
        command: [ "/bin/sh", "-c", "set -e -x; cp -f /etc/kube-flannel/cni-conf.json /etc/cni/net.d/10-flannel.conf; while true; do sleep 3600; done" ]
        volumeMounts:
        - name: cni
          mountPath: /etc/cni/net.d
        - name: flannel-cfg
          mountPath: /etc/kube-flannel/
      volumes:
        - name: run
          hostPath:
            path: /run
        - name: cni
          hostPath:
            path: /etc/cni/net.d
        - name: flannel-cfg
          configMap:
            name: kube-flannel-cfg
```

Flannel rbac manifest

```
---
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: flannel
rules:
  - apiGroups:
      - ""
    resources:
      - pods
    verbs:
      - get
  - apiGroups:
      - ""
    resources:
      - nodes
    verbs:
      - list
      - watch
  - apiGroups:
      - ""
    resources:
      - nodes/status
    verbs:
      - patch
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: flannel
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: flannel
subjects:
- kind: ServiceAccount
  name: flannel
  namespace: kube-system
``` 

Interface state

```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: ens3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 52:54:00:fb:8b:d2 brd ff:ff:ff:ff:ff:ff
    inet 10.0.11.10/24 brd 10.0.11.255 scope global ens3
       valid_lft forever preferred_lft forever
6: docker0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default
    link/ether 02:42:c2:a5:ef:a1 brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/24 scope global docker0
       valid_lft forever preferred_lft forever
7: flannel.1: <BROADCAST,MULTICAST> mtu 1450 qdisc noop state DOWN group default
    link/ether 62:92:55:65:70:e4 brd ff:ff:ff:ff:ff:ff
```

**Comment by strigona-worksight** [comment]
I just got this working myself.

Download kube-flannel.yml and update references to the flannel image from 0.7.0 to 0.7.1.

Re-run the configuration from scratch, but use
`kubectl apply -f kube-flannel.yml`
 (your edited file)

**Comment by strigona-worksight** [comment]
This should be fixed now #690

**Comment by vjdhama** [comment]
@strigona-worksight Thank you very much. The above works.
